functions/BBoxCombFunction.cc

```diff
@@ -0,0 +1,136 @@
+// -*- mode: c++; c-basic-offset:4 -*-
+
+// This file is part of bes, A C++ implementation of the OPeNDAP
+// Hyrax data server
+
+// Copyright (c) 2015 OPeNDAP, Inc.
+// Authors: James Gallagher <jgallagher@opendap.org>
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+//
+// You can contact OPeNDAP, Inc. at PO Box 112, Saunderstown, RI. 02874-0112.
+
+#include "config.h"
+
+#include <cassert>
+#include <sstream>
+#include <memory>
+
+#include <BaseType.h>
+#include <Int32.h>
+#include <Str.h>
+#include <Array.h>
+#include <Structure.h>
+
+#include <D4RValue.h>
+#include <Error.h>
+#include <debug.h>
+#include <util.h>
+#include <ServerFunctionsList.h>
+
+#include "BBoxCombFunction.h"
+#include "roi_util.h"
+
+using namespace std;
+using namespace libdap;
+
+namespace functions {
+
+/**
+ * @brief Combine two bounding boxes with different shapes, forming their combination.
+ *
+ * This combines 2 BBox variables (Array of Structure) forming
+ * their combination.
+ *
+ * @note There is DAP2 versions of this function.
+ *
+ * @param argc Argument count - must be 2
+ * @param argv Argument vector - variable in the current DDS
+ * @param dds The current DDS
+ * @param btpp Value-result parameter for the resulting Array of Structure
+ */
+
+void
+function_dap2_bbox_comb(int argc, BaseType *argv[], DDS &, BaseType **btpp)
+{
+    const string wrong_args = "Wrong number of arguments to bbox_comb(). Expected two bounding boxes";
+
+    unsigned int rank = 0;
+    unsigned int rnk1= 0;
+    unsigned int rnk2 = 0;
+
+    switch (argc) {
+    case 0:
+    case 1:
+        // Must have 2 arguments
+        throw Error(malformed_expr, wrong_args);
+
+    default:
```

Comment:
Use 'case 2:' and not default and have default: throw an exception. As the code is now, the number of args can be 3, 4, ..., although only the first two will be used.